### ISSUE TITLE
Use `void` for empty argument lists in C

### ISSUE THREAD

**Issue by szhorvat** [igraph/stimulus#6]
`void f()` is not a valid function prototype in C for empty argument lists. Stimulus should generate `void f(void)` in these cases. This is important for R to avoid warnings that CRAN cares about.

**Comment by ntamas** [comment]
Should be fixed now (see last two commits).

**Comment by szhorvat** [comment]
Hm, this still seems to generate `R_igraph_version()` instead of `R_igraph_version(void)`.

But fixing this is not urgent since the affected functions aren't actually used at this moment. Just enjoy your trip :-)

**Comment by ntamas** [comment]
Which file is that?  I've found `SEXP R_igraph_version() {` in `src/rinterface.c` but here it is valid to omit `void`, right?

**Comment by szhorvat** [comment]
> but here it is valid to omit `void`, right?

No, it is not, as there was no other prototype, and with `()` the argument list is left unspecified. It should be `(void)`.

In C++, `foo()` means no arguments. In C, `foo()` means no information provided about the arguments. Not providing any argument info is "deprecated", according to the compiler.

```
warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
int foo() { return 1; }
       ^
        void
```

**Comment by ntamas** [comment]
Okay, this should now be fixed in 0.21.4.
